VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts

```diff
@@ -34,20 +37,45 @@ describe('SelectHearingLayoutComponent', () => {
         videoCallService.getPreferredLayout.and.returnValue(layout);
         component.ngOnInit();
         expect(component.selectedLayout).toBe(layout);
+        expect(component.currentButtonContentKey).toBe(buttonContentKeyWhenOpen);
+    });
+
+    it('should call translate service to update accordion button when language changes', () => {
+        const expectedTranslatedContentForButton = 'this is translated for open all button';
+        component.currentButtonContentKey = buttonContentKeyWhenOpen;
+        translateServiceSpy.instant
+            .withArgs(`select-hearing-layout.${component.currentButtonContentKey}`)
+            .and.returnValue(expectedTranslatedContentForButton);
+        component.ngOnInit();
+        onLangChangeSpy.emit({ lang: 'tl' } as LangChangeEvent);
+        expect(component.accordionOpenAllElement.innerHTML).toContain(expectedTranslatedContentForButton);
+        expect(component.currentButtonContentKey).toBe(buttonContentKeyWhenOpen);
+    });
+
+    it('should call translate service to update accordion header when language changes', () => {
+        const expectedTranslatedContentForHeader = 'this is translated for the accordion header';
+        translateServiceSpy.instant
+            .withArgs('select-hearing-layout.choose-hearing-layout')
+            .and.returnValue(expectedTranslatedContentForHeader);
+        component.ngOnInit();
+        onLangChangeSpy.emit({ lang: 'tl' } as LangChangeEvent);
+        expect(headingButton.innerHTML).toContain(expectedTranslatedContentForHeader);
```

Comment:
**TLDR: Tests should not test things that are outside the scope of the test name.**

Minor change. Can the names of these tests be changed?

Here we are testing that the translate service is called, but actually it is going above and beyond that and also testing that the UI is updated with the translated text.

In an ideal unit test world, we should have two unit tests, one that checks that the translate service is called and another that checks the UI shows the translated text.

I don't particularly care about the splitting out of the unit tests as there are pros and cons for that, but if we are combining tests like we are here, then the name of the test should be the most encompassing. Something like, "should show translated text on open accordion button". That way, the test for calling the translate service can be inferred from the test name.

Reply:
- Initially was using TDD to force me to add that property
- atleast from what I have read from jasmine and spy, using line 46-49 would return an expected value when its called with the args passed there, unless I am wrong?

Reply:
I'll give you a call as I think my explanation is lacking 😂